cmd/builder/internal/builder/main.go

```diff
@@ -110,6 +130,16 @@ func Compile(cfg Config) error {
 
 	cfg.Logger.Info("Compiling")
 
+	env := []string{}
+	if !cfg.Distribution.CgoEnabled {
+		initialEnv, err := getGoEnv(cfg)
+		if err != nil {
+			return err
+		}
+		env = append(env, initialEnv...)
```

Comment:
I'm not a huge fan of having this here, but when setting `CGO_ENABLED=0` without it, I get this locally:
```
exit status 1, error message: warning: GOPATH set to GOROOT () has no effect
go: module cache not found: neither GOMODCACHE nor GOPATH is set
```

Looking at the output of `go env` when the env var is set vs. when it isn't, the output is different. So I just got the output when the env var is unset and copy it into the variables for when we set it. This feels off, but at the same time if the `go` cli is changing other variables in unexpected ways, perhaps it's safer for us to explicitly set these.